### ISSUE TITLE
Can't Update Driver After Upgrading Linux Kernel

### ISSUE THREAD

**Issue by computer-science-student** [tomaspinho/rtl8821ce#309]
This driver use to work for me, but ever since Debian upgraded my kernel, the Wifi driver no longer works. I run Debian with linux kernel 5.10.0-18-amd64.
I added this code:
blacklist rtw88_8821ce
to the file /etc/modprobe.d/blacklist.conf
I uninstalled the driver, pulled the git repository, make clean, and tried to reinstall the driver, but I am unable to install the driver again.
This is what happens when I try to install the driver:

`
$ sudo ./dkms-install.sh 
About to run dkms install steps...

Creating symlink /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/source ->
                 /usr/src/rtl8821ce-v5.5.2_34066.20200325

DKMS: add completed.

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...(bad exit status: 2)
...(bad exit status: 2)18-amd64
Error! Bad return status for module build on kernel: 5.10.0-18-amd64 (x86_64)
Consult /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/make.log for more information.

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...(bad exit status: 2)
...(bad exit status: 2)18-amd64
Error! Bad return status for module build on kernel: 5.10.0-18-amd64 (x86_64)
Consult /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/make.log for more information.
Finished running dkms install steps.`

How do I get this driver working again? Please advise. Thanks!

**Comment by ramazanemreosmanoglu** [comment]
I had the same issue twice, on endeavour and garuda. Kernel upgrade happens automatically on arch and arch-based distributions. Using a non-rolling distribution might help for a stable system for now.

**Comment by computer-science-student** [comment]
Were you able to get this working after having the same issue? I have this problem on Debian, a non-rolling distribution. I can't even install the driver any more with sudo ./dkms-install.sh

**Comment by tomaspinho** [comment]
Can you get us the contents of the file at `/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/make.log`?

**Comment by ramazanemreosmanoglu** [comment]
> Were you able to get this working after having the same issue? I have this problem on Debian stable, a non-rolling distribution. I can't even install the driver any more with sudo ./dkms-install.sh because the installer says "Error! Bad return status for module build on kernel: 5.10.0-18-amd64 (x86_64)".

Well, I gave devuan a try for a week and didn't like it. I didn't use it for a long time to test this.

But, that's really weird. Debian stable doesn't actually update the kernel unless you say so. Are you sure there were no such an update about kernel stuff?

I currently use Arch, added ``linux-lts`` and ``linux-lts-headers`` to the ``pacman.conf`` IgnorePkg section to disable updates for those packages. I will report here if I have issues.

**Comment by computer-science-student** [comment]
tomaspinho, the contents of make.log are as follows:

DKMS make.log for rtl8821ce-v5.5.2_34066.20200325 for kernel 5.10.0-18-amd64 (x86_64)
Mon 02 Jan 2023 07:47:10 AM CST
/build M=/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build  modules
/build: No such file or directory.  Stop.
make: *** [Makefile:2244: modules] Error 2

**Comment by computer-science-student** [comment]
ramazanemreosmanoglu,

I do think that an update to a new kernel version is the cause for why my driver broke. Here is why.

When I had an earlier kernel version, this driver worked for me, and then after one sudo apt-get update && sudo apt-get upgrade update later it broke. For a couple of months, I could manually go back to a previous kernel version (version 4 or around there) in GRUB every time I booted, but now another update got rid of that kernel version in grub, and I can no longer do that.
So I do think that the update to a new kernel version is the cause for why my driver broke.

**Comment by ramazanemreosmanoglu** [comment]
> ramazanemreosmanoglu,
> 
> I do think that an update to a new kernel version is the cause for why my driver broke. Here is why.
> 
> When I had an earlier kernel version, this driver worked for me, and then after one sudo apt-get update && sudo apt-get upgrade update later it broke. For a couple of months, I could manually go back to a previous kernel version (version 4 or around there) in GRUB every time I booted, but now another update got rid of that kernel version in grub, and I can no longer do that. So I do think that the update to a new kernel version is the cause for why my driver broke.

https://www.snel.com/support/how-to-block-package-and-kernel-updates-in-debian/

I currently use debian too. You can tell apt to not to update kernel if you use generic kernel for now.

**Comment by a-h-ismail** [comment]
I don't think that the newer kernels are an issue. I have this driver on Fedora 37 (kernel 6.1) and it works as well as it did on Kernel 5.4.
